src/viking.js

```diff
@@ -1,11 +1,80 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength){
+        this.health = health;
+        this.strength = strength;
+    }
+
+    attack(){
+        return this.strength
+    }
+
+    receiveDamage(damage){
+        this.health = this.health - damage;
+    }
+
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength){
+        super(health, strength);
+        this.name = name;
+    }
+
+    receiveDamage(damage){
+        this.health = this.health - damage;
+        if (this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`;
+        } else if (this.health <= 0){
+            return `${this.name} has died in act of combat`;
+        }  
+    }
+
+    battleCry(){
+        return "Odin Owns You All!";
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    
+    receiveDamage(damage){
+        this.health = this.health - damage;
+        if (this.health > 0) {
+            return `A Saxon has received ${damage} points of damage`;
+        } else if (this.health <= 0) {
+            return `A Saxon has died in combat`;
+        }   
+    }
+}
 
 // War
-class War {}
+class War {
+    constructor(){
+        this.vikingArmy = [];
+        this.saxonArmy = [];
+    }
+
+    addViking(Viking){
+        this.vikingArmy.push(Viking);
+    }
+
+    addSaxon(Saxon){
+        this.saxonArmy.push(Saxon);
+    }
+
+
+    //This is where I got stuck :(
```

Comment:
This is a challenging part of the assignment!

Let me help you in pseudocode. What we would like to do is the following:
- Select a (random) viking from the vikingArmy
- Select a (random) Saxon from the saxonArmy
- Utilize the viking's receiveDamage method and the Saxon's attack method to subtract health points from the viking's health
- If the viking has no health points left, it should be removed from the vikingArmy array